codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/generators/UnionGenerator.kt

```diff
@@ -24,12 +28,17 @@ class UnionGenerator(
     private val writer: RustWriter,
     private val shape: UnionShape
 ) {
+    private val sortedMembers: List<MemberShape> = shape.allMembers.values
```

Comment:
hmm...maybe we should handle this as a model transformation? Generally I try to keep this stuff out of the generators

Reply:
(but if you need it in the model for later, I understand.)

Reply:
EventStreaming is a protocol level detail...it's possible that this union could be used somewhere else. Probably worth thinking about this more.